theano/sparse/sandbox/sp2.py

```diff
@@ -116,6 +117,11 @@ def make_node(self, n, p, shape):
         n = tensor.as_tensor_variable(n)
         p = tensor.as_tensor_variable(p)
         shape = tensor.as_tensor_variable(shape)
+
+        assert n.dtype in discrete_dtypes
+        assert p.dtype not in discrete_dtypes
```

Comment:
yeah..so, change "assert p.dtype not in discrete_dtypes" to "assert p.dtype in float_dtypes" ?